File: exercism_data/src/571.rb
require 'pry'
class Phrase
  attr_accessor :words

  def initialize(words)
    @words = words.split(/\W/)
                  .map(&:downcase)
                  .reject(&:empty?)
  end

  def word_count
    words.group_by { |word| words.find { |el| el == word } }
         .map { |k,v| { k => v.count } }
         .inject(&:merge)
  end
end

File: assignments/python/wc/src/571.py
def word_count(word):
    if isinstance(word, basestring) and len(word) > 0:
        words = word.split()
        counts = {}
        for a in words:
            if a in counts:
                counts[a] += 1
            else:
                counts[a] = 1
        return counts
